src/communication/v2/listener.hpp

```diff
@@ -68,33 +76,63 @@ class Listener final : public std::enable_shared_from_this<Listener<TSession, TS
         endpoint_{endpoint},
         service_name_{service_name},
         inactivity_timeout_{inactivity_timeout_sec} {
+    TryCreate<false>();
+  }
+
+  Listener(assert_create /*obj*/, boost::asio::io_context &io_context, TSessionContext *session_context,
+           ServerContext *server_context, tcp::endpoint &endpoint, const std::string_view service_name,
+           const uint64_t inactivity_timeout_sec)
+      : io_context_(io_context),
+        session_context_(session_context),
+        server_context_(server_context),
+        acceptor_(io_context_),
+        endpoint_{endpoint},
+        service_name_{service_name},
+        inactivity_timeout_{inactivity_timeout_sec} {
+    TryCreate<true>();
+  }
```

Comment:
@Ignition I tried to make ctor be `template<bool AssertCreate>` but then I had some template issues and didn't input more time into this, let me know if you have any idea how to make this better